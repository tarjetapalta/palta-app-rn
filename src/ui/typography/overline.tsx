import React, { FunctionComponent } from 'react';

import { Text, TextProps } from './base';

export const Overline: FunctionComponent<TextProps> = ({
  /** Test id for test */
  testID = 'Overline',
  /** Text */
  children,
  /** Text color */
  color,
  /** Text font opacity */
  emphasis,
  /** Text align */
  textAlign,
}) => (
  <Text
    testID={testID}
    color={color}
    size="xxs"
    weight="light"
    emphasis={emphasis}
    textAlign={textAlign}
  >
    {children}
  </Text>
);
