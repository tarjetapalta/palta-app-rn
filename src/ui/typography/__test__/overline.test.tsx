import React from 'react';
import renderer from 'react-test-renderer';

import WithTheme from '../../../theme/context';
import { Overline } from '../overline';

describe('overline component', () => {
  it('basic Overline with color "text", emphasis "high" and testID "Overline"', () => {
    const result = renderer
      .create(
        <WithTheme>
          <Overline>Some text</Overline>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
  it('basic Overline with color "primary", emphasis "medium"', () => {
    const result = renderer
      .create(
        <WithTheme>
          <Overline color="primary" emphasis="medium">
            Some text
          </Overline>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
  it('basic with custom testId', () => {
    const result = renderer
      .create(
        <WithTheme>
          <Overline testID="custom testID">Some text</Overline>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
});
