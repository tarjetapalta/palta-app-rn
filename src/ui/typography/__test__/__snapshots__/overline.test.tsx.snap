// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overline component basic Overline with color "primary", emphasis "medium" 1`] = `
<Text
  color="primary"
  emphasis="medium"
  numberOfLines={0}
  size="xxs"
  style={
    Array [
      Object {
        "color": "#91b086",
        "fontFamily": "BwModelica-Light",
        "fontSize": 10,
        "opacity": 0.6,
        "textAlign": "center",
      },
    ]
  }
  testID="Overline"
  weight="light"
>
  Some text
</Text>
`;

exports[`overline component basic Overline with color "text", emphasis "high" and testID "Overline" 1`] = `
<Text
  numberOfLines={0}
  size="xxs"
  style={
    Array [
      Object {
        "color": "#283238",
        "fontFamily": "BwModelica-Light",
        "fontSize": 10,
        "opacity": 0.87,
        "textAlign": "center",
      },
    ]
  }
  testID="Overline"
  weight="light"
>
  Some text
</Text>
`;

exports[`overline component basic with custom testId 1`] = `
<Text
  numberOfLines={0}
  size="xxs"
  style={
    Array [
      Object {
        "color": "#283238",
        "fontFamily": "BwModelica-Light",
        "fontSize": 10,
        "opacity": 0.87,
        "textAlign": "center",
      },
    ]
  }
  testID="custom testID"
  weight="light"
>
  Some text
</Text>
`;
